### PR TITLE
PM-11248: Add isUsingKeyConnector flag to UserState

### PR DESCRIPTION
**Pull request by david-livefront**
## 🎟️ Tracking

[PM-11248](https://bitwarden.atlassian.net/browse/PM-11248)

## 📔 Objective

This PR adds the `isUsingKeyConnector` flag to the `UserState`, this will be needed for state-based navigation when displaying and dismissing the `RemovePasswordScreen`.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-11248]: https://bitwarden.atlassian.net/browse/PM-11248?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryImpl.kt [hunk 1]
```diff
@@ -238,6 +241,7 @@ class AuthRepositoryImpl(
         authDiskSource.userStateFlow,
         authDiskSource.userAccountTokensFlow,
         authDiskSource.userOrganizationsListFlow,
+        authDiskSource.userIsUsingKeyConnectorFlow,
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryImpl.kt:244]: This is more a general question for this block of code, is there any reason to not just add this at the end to keep the pre existing order in the subsequent lines?

Comment (david-livefront) [reply app/src/main/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryImpl.kt:244]: I was attempting to keep all the `authDiskSource` stuff together.

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryImpl.kt:244]: works for me!

File: app/src/main/java/com/x8bit/bitwarden/data/auth/repository/util/AuthDiskSourceExtensions.kt [hunk 2]
```diff
@@ -100,6 +101,47 @@ val AuthDiskSource.userAccountTokensFlow: Flow<List<UserAccountTokens>>
         }
         .distinctUntilChanged()
 
+/**
+ * Returns the current list of [UserKeyConnectorState].
+ */
+val AuthDiskSource.userIsUsingKeyConnectorsList: List<UserKeyConnectorState>
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/auth/repository/util/AuthDiskSourceExtensions.kt:107]: ⛏️ Should we rename these too, sorry for not calling this out with the other comment. I am not sure if I feel like this is too much of a nit.

Comment (david-livefront) [reply app/src/main/java/com/x8bit/bitwarden/data/auth/repository/util/AuthDiskSourceExtensions.kt]: Nope, that's a good change and I am on it!

**Review by dseverns-livefront — APPROVED** [review]
NOICE!

**Comment by david-livefront** [comment]
Thanks @dseverns-livefront

**Pull request merged**